lib/msf/core/post_mixin.rb

```diff
@@ -183,91 +185,37 @@ def session_incompatibility(sess_or_sid)
       s = sess_or_sid
     end
 
+    issues = []
+
     # Can't do anything without a session
-    return 'invalid session' unless s
+    unless s
+      issues << ['invalid session']
+      return issues
+    end
 
     # Can't be compatible if it's the wrong type
     if session_types
-      return "incompatible session type: #{s.type}" unless session_types.include?(s.type)
+      issues << "incompatible session type: #{s.type}" unless session_types.include?(s.type)
     end
 
     # Check to make sure architectures match
     mod_arch = self.module_info['Arch']
     if mod_arch
       mod_arch = [mod_arch] unless mod_arch.kind_of?(Array)
       # Assume ARCH_CMD modules can work on supported SessionTypes since both shell and meterpreter types can execute commands
-      return "incompatible session architecture: #{s.arch}" unless mod_arch.include?(s.arch) || mod_arch.include?(ARCH_CMD)
+      issues << "incompatible session architecture: #{s.arch}" unless mod_arch.include?(s.arch) || mod_arch.include?(ARCH_CMD)
     end
 
     # Arch is okay, now check the platform.
     if self.platform && self.platform.kind_of?(Msf::Module::PlatformList)
-      return "incompatible session platform: #{s.platform}" unless self.platform.supports?(Msf::Module::PlatformList.transform(s.platform))
+      issues << "incompatible session platform: #{s.platform}" unless self.platform.supports?(Msf::Module::PlatformList.transform(s.platform))
```

Comment:
I wanted to return the potential platform issue, as well as the meterpreter issues, so returning early wasn't the right pattern here for me.

Example:

```
[!] SESSION may not be compatible with this module (incompatible session platform: linux. missing Meterpreter features: stdapi_ui_desktop_screenshot, stdapi_ui_send_keyevent, stdapi_ui_send_mouse, espia_image_get_dev_screen)
[*] Using URL: http://127.0.0.1:8080/WNOheBiKs
[*] Server started.
```